src/envs/cover.py

```diff
@@ -1089,3 +1206,6 @@ def _create_initial_state(self, rng: np.random.Generator) -> State:
         del rng
         zero_rng = np.random.default_rng(0)
         return super()._create_initial_state(zero_rng)
+
+class MaxPlacementsFailure(Exception):
```

Comment:
if we aren't planning to catch this kind of exception specifically, I think we can just remove it and use `Exception` or `RuntimeError`